### PR TITLE
[FEATURE] Automatic Aquarium Feeder Integration: Add feeder support with servo-based feeding mechanism

### PR DESCRIPTION
**Pull request by ryan-gomes-8178**
## Description

This PR introduces a new **Feeder** subsystem to TerrariumPI that enables automatic feeding of aquarium inhabitants using a servo-controlled sliding mechanism.

### What's New

- **Feeder Entity**: New database model for managing automatic feeders
- **Servo Driver**: Hardware abstraction for controlling SG90 servo motors
- **Scheduling**: Integrated with TerrariumPI's scheduling system (e.g., morning/night feedings)
- **REST API**: Complete API for managing feeders and feeding schedules
- **Web UI**: New Feeders page with visual management interface
- **Feeding History**: Automatic logging of all feeding events for monitoring

### Use Case

Users can now set up one or more automatic aquarium feeders that:
- Dispense food at configurable times (morning, night, custom)
- Control portion sizes per feeding
- Track feeding history and patterns
- Manually trigger feeding on-demand
- Test servo movement before actual feeding

---

### Architecture

The feature follows TerrariumPI's established patterns:

1. **Database Layer** (`terrariumDatabase.py`):
   - `Feeder` entity for feeder configuration
   - `FeedingHistory` entity for event logging

2. **Hardware Layer** (`hardware/feeder/__init__.py`):
   - `terrariumFeeder` class for servo control
   - PWM-based angle positioning (0-180°)
   - Timed feed sequences with configurable timing

3. **Engine Layer** (`terrariumEngine.py`):
   - Feeder instance management
   - Schedule checking and automatic triggering
   - Callback system for status updates

4. **API Layer** (`terrariumAPI.py`):
   - RESTful endpoints for CRUD operations
   - Manual feed triggering
   - History export (CSV)

5. **Frontend Layer** (`gui/pages/Feeders.svelte`):
   - Feeder management dashboard
   - Schedule configuration UI
   - Real-time feeding trigger and testing

### Key Design Decisions

**Why a new Feeder entity instead of extending Relay?**
- Servo requires precise angle positioning, not binary ON/OFF
- Feeding requires sequential timed operations (rotate → hold → return)
- Natural fit with area-based scheduling
- Extensible for future multi-feeder scenarios

**Why PWM instead of direct GPIO control?**
- SG90 servo requires PWM signal (50Hz frequency)
- Existing PWMOutputDevice infrastructure in gpiozero
- Precise timing control and angle mapping

**Why schedule in JSON instead of separate cron table?**
- Aligns with TerrariumPI's existing patterns (Areas, relay schedules)
- Per-feeder schedule isolation
- Easy to extend with additional feed times

---

### New Files
- `migrations/002_add_feeder_support.py` - Database schema
- `hardware/feeder/__init__.py` - Feeder hardware driver
- `gui/pages/Feeders.svelte` - Main feeder management page
- `gui/components/feeders/FeedersCard.svelte` - Feeder display card
- `gui/components/feeders/FeedersForm.svelte` - Feeder editor form
- `gui/components/feeders/FeedersSchedule.svelte` - Schedule management (optional)

### Modified Files
- `terrariumDatabase.py`:
  - Added `Feeder` ORM entity
  - Added `FeedingHistory` ORM entity
  - Default settings updated

- `terrariumEngine.py`:
  - Added `self.feeders` dictionary
  - Added `load_feeders()` method
  - Added `callback_feeder()` method
  - Added `check_feeder_schedules()` method
  - Added feeder schedule checking to main loop

- `terrariumAPI.py`:
  - Added 8 new routes under `/api/feeders/*`
  - Added corresponding handler methods
  - No changes to existing routes

- `gui/pages/index.js`:
  - Added import for Feeders page

---

### Feeder Management
```
GET    /api/feeders/                    - List all feeders
POST   /api/feeders/                    - Create new feeder
GET    /api/feeders/<id>/               - Get feeder detail
PUT    /api/feeders/<id>/               - Update feeder
DELETE /api/feeders/<id>/               - Delete feeder
```

### Feeder Operations
```
POST   /api/feeders/<id>/feed/          - Trigger immediate feeding
POST   /api/feeders/<id>/test/          - Test servo movement
GET    /api/feeders/<id>/history/       - Get feeding history (default: last day)
GET    /api/feeders/<id>/history/<period>/  - Get history for period (hour|day|week|month|year)
```

### Request/Response Examples

**Create Feeder**
```json
POST /api/feeders/
{
  "enclosure": "enc-123",
  "name": "Main Tank Feeder",
  "hardware": "17",
  "enabled": true,
  "servo_config": {
    "feed_angle": 90,
    "rest_angle": 0,
    "rotate_duration": 1000,
    "feed_hold_duration": 1500,
    "portion_size": 1.0
  },
  "schedule": {
    "morning": {
      "time": "08:00",
      "enabled": true,
      "portion_size": 1.0
    },
    "night": {
      "time": "20:00",
      "enabled": true,
      "portion_size": 1.0
    }
  }
}
```

**Manual Feed**
```json
POST /api/feeders/feed-123/feed/
{
  "portion_size": 1.5
}

Response:
{
  "status": "success",
  "message": "Feeder 'Main Tank Feeder' completed feed sequence successfully",
  "timestamp": 1705843200.0,
  "portion_size": 1.5
}
```

**Get Feeding History**
```json
GET /api/feeders/feed-123/history/day/

Response:
{
  "data": [
    {
      "timestamp": 1705824000.0,
      "status": "success",
      "portion_size": 1.0
    },
    {
      "timestamp": 1705867200.0,
      "status": "success",
      "portion_size": 1.0
    }
  ]
}
```

---

### Servo Configuration Parameters

```javascript
servo_config: {
  feed_angle: 90,              // Angle to open mechanism (degrees, 0-180)
  rest_angle: 0,              // Angle to close mechanism (degrees, 0-180)
  rotate_duration: 1000,      // Time to rotate (milliseconds)
  feed_hold_duration: 1500,   // Time to hold at feed position (milliseconds)
  portion_size: 1.0           // Default portion per feeding (grams/units)
}
```

### Schedule Configuration

```javascript
schedule: {
  morning: {
    time: "08:00",           // HH:MM format, 24-hour
    enabled: true,           // Whether this schedule is active
    portion_size: 1.0        // Portion size for this feeding
  },
  night: {
    time: "20:00",
    enabled: true,
    portion_size: 1.0
  }
}
```

### GPIO Pin Requirements

- Use any GPIO pin that supports PWM output
- No conflicts with other hardware (sensors, relays, buttons)
- Common choices: GPIO 17, 18, 27, 22
- Power requirements: 5V for servo power, GPIO logic levels (3.3V)

### Hardware Wiring

```
SG90 Servo:
- Brown wire  → GND (Pin 6, 9, 14, 20, 25, 30, 34, 39)
- Red wire    → 5V (Pin 2, 4)
- Orange wire → GPIO (user specified, e.g., GPIO 17 = Pin 11)
```

---

### Pre-Testing Checklist
- [ ] Raspberry Pi is running and accessible
- [ ] SG90 servo is properly wired to GPIO pin
- [ ] Servo has adequate 5V power supply
- [ ] TerrariumPI database is backed up
- [ ] No other services using the selected GPIO pin

### Unit Testing

1. **Database Migration Test**
   ```bash
   # Run migration
   python -c "from terrariumDatabase import init; init('test')"
   # Verify tables exist:
   sqlite3 data/terrariumpi.db ".tables" | grep feeder
   ```

2. **Hardware Driver Test**
   ```bash
   # Create a simple test script
   python test_feeder.py
   ```

### Integration Testing

1. **API Endpoint Test**
   ```bash
   # Get enclosures to use in feeder creation
   curl -X GET http://localhost:8090/api/enclosures/ -H "Authorization: Bearer TOKEN"
   
   # Create a test feeder
   curl -X POST http://localhost:8090/api/feeders/ \
     -H "Authorization: Bearer TOKEN" \
     -H "Content-Type: application/json" \
     -d '{
       "enclosure": "ENC-ID-HERE",
       "name": "Test Feeder",
       "hardware": "17",
       "servo_config": {...}
     }'
   ```

2. **Manual Feed Test**
   ```bash
   curl -X POST http://localhost:8090/api/feeders/FEEDER-ID/feed/ \
     -H "Authorization: Bearer TOKEN" \
     -d '{"portion_size": 1.0}'
   
   # Observe servo should:
   # 1. Rotate to feed position (~1 second)
   # 2. Hold at feed position (~1.5 seconds)
   # 3. Return to rest position (~1 second)
   ```

3. **Servo Test Endpoint**
   ```bash
   curl -X POST http://localhost:8090/api/feeders/FEEDER-ID/test/ \
     -H "Authorization: Bearer TOKEN"
   
   # Same movement pattern as feed, but doesn't record history
   ```

4. **Web UI Test**
   - Navigate to http://localhost:8090
   - New "Feeders" page should appear in navigation
   - Can create feeder without errors
   - Can view feeder details
   - Can manually trigger feed from UI
   - Servo performs feeding sequence

5. **Scheduling Test**
   ```bash
   # Set schedule time to 1 minute in future
   # Wait for scheduled time
   # Observe servo triggers automatically
   # Check /api/feeders/FEEDER-ID/history/ shows entry
   ```

6. **History Export Test**
   ```bash
   curl -X GET http://localhost:8090/api/feeders/FEEDER-ID/history/day/ \
     -H "Authorization: Bearer TOKEN"
   
   # Should show CSV format with timestamp|status|portion_size
   ```

### Edge Cases to Test

- [ ] Multiple feeders on different GPIO pins
- [ ] Disable schedule and verify no auto-feeding
- [ ] Edit feeder while TerrariumPI is running
- [ ] Delete feeder - ensure hardware is cleanly stopped
- [ ] GPIO pin in use by another device - should error gracefully
- [ ] Servo not powered - should timeout gracefully
- [ ] Change schedule time and verify new time triggers
- [ ] Export history to CSV and verify format

### Regression Testing

Ensure existing features still work:
- [ ] Relays still toggle correctly
- [ ] Sensors still read values
- [ ] Buttons still register presses
- [ ] Webcams still capture images
- [ ] Calendar events still trigger
- [ ] Areas/enclosures still function
- [ ] Authentication still works
- [ ] Database backups still work

---

## Rollback Instructions

If issues occur after deployment:

1. **Disable feeders temporarily**
   ```bash
   # Comment out feeder loading in terrariumEngine.py:
   # self.load_feeders()  # Temporarily disabled
   
   # Restart TerrariumPI
   ```

2. **Full rollback (preserve data)**
   ```bash
   # Backup current database
   cp data/terrariumpi.db data/terrariumpi.db.backup
   
   # Revert all feeder files and code changes
   git revert <commit-hash>
   
   # Restart TerrariumPI
   ```

3. **Complete removal**
   ```bash
   # Revert feeder-related database changes
   sqlite3 data/terrariumpi.db "DROP TABLE feeder_history; DROP TABLE feeder;"
   
   # Remove feeder code files
   rm -rf hardware/feeder/
   rm gui/pages/Feeders.svelte
   rm -rf gui/components/feeders/
   
   # Restart TerrariumPI
   ```

---

### Logs to Monitor
```bash
tail -f log/terrariumpi.log | grep -i feeder
```

Look for:
- Feeder loading errors on startup
- Schedule execution messages
- Servo movement logs
- Hardware errors

### Performance Notes

- Feeder schedule check runs every 60 seconds (configurable)
- Feeding operation blocks for ~3.5 seconds (rotate + hold + return)
- No impact on other subsystems (separate thread)
- Database queries are indexed by feeder_id

---

### Current Limitations
1. Schedule supports only predefined times (morning/night)
2. No conditional feeding based on sensor values
3. No support for multiple servo angles in sequence
4. Portion sizes are placeholder values (not actual weight measurement)

### Future Enhancements
- [ ] Custom scheduling (cron-like syntax)
- [ ] Sensor-based feeding (feed if water temp > X)
- [ ] Nutrition tracking dashboard
- [ ] Multiple food types/reservoirs
- [ ] Feeding statistics and trends
- [ ] Mobile push notifications on feed
- [ ] AI-driven feeding adjustments
- [ ] Integration with water quality monitoring

---

### First-Time Setup
1. Apply database migration
2. Load TerrariumPI with new code
3. Restart TerrariumPI service
4. Verify feeder page loads in web UI
5. Create first feeder via API or web UI
6. Test manual feeding
7. Verify feeding history recorded

### Upgrade from Previous Version
1. Backup database: `cp data/terrariumpi.db data/terrariumpi.db.bak`
2. Pull new code
3. Restart TerrariumPI (migration runs automatically)
4. Verify no errors in logs
5. Existing feeders (if any) will load automatically

### Production Considerations
- Test servo timing with actual food dispensing
- Adjust `rotate_duration` and `feed_hold_duration` based on mechanical response
- Monitor servo power supply under continuous operation
- Use quality servo with metal gears for long-term reliability
- Consider adding backup power (UPS) for critical feeders

### CODE REVIEW

File: gui/components/feeders/FeedersCard.svelte [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+      isFeedingNow = true;
+      manualFeed(feeder.id, feeder.servo_config.portion_size, (data) => {
+        dispatch('reload');
+        isFeedingNow = false;
+      });
+    } catch (e) {
+      console.error('Feed error:', e);
+      isFeedingNow = false;
+    }
+  }
```
[the remark this answers is not in the document]
Comment (ryan-gomes-8178) [reply gui/components/feeders/FeedersCard.svelte]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/ryan-gomes-8178/PaludariumPI/pull/5#r2715626257)

File: gui/components/feeders/FeedersCard.svelte [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+
+  async function testServo() {
+    try {
+      testFeederServo(feeder.id, (data) => {
+        console.log('Servo test result:', data);
+      });
+    } catch (e) {
+      console.error('Test error:', e);
+    }
+  }
```
[the remark this answers is not in the document]
Comment (ryan-gomes-8178) [reply gui/components/feeders/FeedersCard.svelte]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/ryan-gomes-8178/PaludariumPI/pull/5#r2715626216)

File: gui/components/feeders/FeedersFormModal.svelte [hunk 3]
```diff
@@ -0,0 +1,311 @@
...
+    }).catch(e => {
+      isSaving = false;
+      error = e.message || 'Failed to save feeder';
+      console.error('Form error:', e);
+      errorNotification(
+        error,
+        $_('feeders.menu.title', { default: 'Feeders' })
+      );
+    });
+  }
```
[the remark this answers is not in the document]
Comment (ryan-gomes-8178) [reply gui/components/feeders/FeedersFormModal.svelte:81]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/ryan-gomes-8178/PaludariumPI/pull/5#r2715626247)

File: hardware/feeder/__init__.py [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+            
+            self._device = PWMOutputDevice(
+                bcm_pin,
+                frequency=self.DEFAULT_FREQUENCY,
+                initial_value=0
+            )
+            logger.info(f"Loaded feeder {self.name} on GPIO {gpio_pin}")
+        except Exception as e:
+            logger.error(f"Failed to load feeder hardware: {e}")
+            raise terrariumFeederHardwareException(f"Cannot load GPIO {self.hardware}: {e}")
```
[the remark this answers is not in the document]
Comment (ryan-gomes-8178) [reply hardware/feeder/__init__.py]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/ryan-gomes-8178/PaludariumPI/pull/5#r2715626170)

File: hardware/feeder/__init__.py [hunk 5]
```diff
@@ -0,0 +1,237 @@
...
+                
+            except Exception as e:
+                error_msg = f"Feeder '{self.name}' feed sequence failed: {e}"
+                logger.error(error_msg)
+                
+                # Stop servo movement on error
+                try:
+                    self._device.value = 0
+                except:
+                    pass
```
[the remark this answers is not in the document]
Comment (ryan-gomes-8178) [reply hardware/feeder/__init__.py]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/ryan-gomes-8178/PaludariumPI/pull/5#r2715626333)

File: terrariumAPI.py [hunk 6]
```diff
@@ -1513,6 +1589,7 @@ def relay_history(self, relay, action="history", period="day"):
     def relay_hardware(self):
         return {"data": terrariumRelay.available_relays}
 
+    @orm.db_session(sql_debug=DEBUG, show_values=DEBUG)
     def relay_scan(self):
```
[the remark this answers is not in the document]
Comment (ryan-gomes-8178) [reply terrariumAPI.py:1606]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/ryan-gomes-8178/PaludariumPI/pull/5#r2715626413)

File: terrariumAPI.py [hunk 7]
```diff
@@ -1935,6 +2012,179 @@ def setting_delete(self, setting):
...
+                hardware=request.json["hardware"],
+                servo_config=request.json.get("servo_config", {
+                    "feed_angle": 90,
+                    "rest_angle": 0,
+                    "rotate_duration": 1000,
+                    "feed_hold_duration": 1500,
+                    "portion_size": 1.0
+                }),
+                schedule=request.json.get("schedule", {})
+            )
```
[the remark this answers is not in the document]
Comment (ryan-gomes-8178) [reply terrariumAPI.py:2152]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/ryan-gomes-8178/PaludariumPI/pull/5#r2715626394)

File: terrariumAPI.py [hunk 8]
```diff
@@ -1935,6 +2012,179 @@ def setting_delete(self, setting):
...
+    
+    @orm.db_session(sql_debug=DEBUG, show_values=DEBUG)
+    def feeder_update(self, feeder):
+        from terrariumDatabase import Feeder
+        try:
+            feeder_obj = Feeder[feeder]
+            feeder_obj.name = request.json.get("name", feeder_obj.name)
+            feeder_obj.enabled = request.json.get("enabled", feeder_obj.enabled)
+            feeder_obj.servo_config = request.json.get("servo_config", feeder_obj.servo_config)
+            feeder_obj.schedule = request.json.get("schedule", feeder_obj.schedule)
```
[the remark this answers is not in the document]
Comment (ryan-gomes-8178) [reply terrariumAPI.py]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/ryan-gomes-8178/PaludariumPI/pull/5#r2715626184)

File: terrariumEngine.py [hunk 9]
```diff
@@ -47,6 +47,9 @@
 
 from terrariumNotification import terrariumNotification
 
+from hardware.feeder import terrariumFeeder, terrariumFeederException
```
[the remark this answers is not in the document]
Comment (ryan-gomes-8178) [reply terrariumEngine.py]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/ryan-gomes-8178/PaludariumPI/pull/5#r2715626425)

File: terrariumEngine.py [hunk 10]
```diff
@@ -2038,3 +2047,89 @@ def total_power_and_water_usage(self, force=False, background=False, thread_retu
...
+        """Callback when feeder operation completes"""
+        from terrariumDatabase import Feeder, FeedingHistory
+        
+        @orm.db_session
+        def _update():
+            try:
+                feeder = Feeder[feeder_id]
+                FeedingHistory(
+                    feeder=feeder,
+                    timestamp=datetime.now(),
```
[the remark this answers is not in the document]
Comment (ryan-gomes-8178) [reply terrariumEngine.py]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/ryan-gomes-8178/PaludariumPI/pull/5#r2715626234)

File: terrariumEngine.py [hunk 11]
```diff
@@ -2038,3 +2047,89 @@ def total_power_and_water_usage(self, force=False, background=False, thread_retu
...
+                    current_time = now.strftime("%H:%M")
+                    
+                    for feed_name, feed_config in schedule.items():
+                        if not feed_config.get('enabled', True):
+                            continue
+                        
+                        if feed_config.get('time') == current_time:
+                            # Check if we already fed in this minute
+                            last_history = feeder_db.history.filter(
+                                lambda h: h.timestamp >= now - timedelta(minutes=1)
```
[the remark this answers is not in the document]
Comment (ryan-gomes-8178) [reply terrariumEngine.py]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/ryan-gomes-8178/PaludariumPI/pull/5#r2715626207)

File: terrariumEngine.py [hunk 12]
```diff
@@ -2038,3 +2047,89 @@ def total_power_and_water_usage(self, force=False, background=False, thread_retu
...
+                            ).order_by(orm.desc(FeedingHistory.timestamp)).first()
+                            
+                            if not last_history:
+                                portion = feed_config.get('portion_size', feeder_db.servo_config.get('portion_size', 1.0))
+                                # Run in thread to avoid blocking
+                                threading.Thread(
+                                    target=feeder.feed,
+                                    args=(portion,),
+                                    daemon=True
+                                ).start()
```
[the remark this answers is not in the document]
Comment (ryan-gomes-8178) [reply terrariumEngine.py]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/ryan-gomes-8178/PaludariumPI/pull/5#r2715626187)

File: terrariumEngine.py [hunk 13]
```diff
@@ -2029,12 +2045,133 @@ def total_power_and_water_usage(self, force=False, background=False, thread_retu
...
+                                .order_by(orm.desc(FeedingHistory.timestamp))
+                                .first()
+                            )
+
+                            if not last_history:
+                                portion = feed_config.get(
+                                    "portion_size", feeder_db.servo_config.get("portion_size", 1.0)
+                                )
+                                # Run in thread to avoid blocking
+                                threading.Thread(target=feeder.feed, args=(portion,), daemon=True).start()
```
[the remark this answers is not in the document]
Comment (ryan-gomes-8178) [reply terrariumEngine.py]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/ryan-gomes-8178/PaludariumPI/pull/5#discussion_r2716274007)

[the remark this answers is not in the document]
Comment (ryan-gomes-8178) [reply terrariumEngine.py]: will be addressed as a follow-up PR

File: terrariumDatabase.py [hunk 14]
```diff
@@ -680,3 +681,59 @@ def to_dict(self, only=None, exclude=None, with_collections=False, with_lazy=Fal
...
+        return data
+    
+    def __repr__(self):
+        return f"Feeder '{self.name}' for enclosure '{self.enclosure.name}'"
+
+
+class FeedingHistory(db.Entity):
+    """Record of each feeding event"""
+    
+    feeder = orm.Required("Feeder")
```
[the remark this answers is not in the document]
Comment (ryan-gomes-8178) [reply terrariumDatabase.py:731]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/ryan-gomes-8178/PaludariumPI/pull/5#discussion_r2716274057)

[the remark this answers is not in the document]
Comment (ryan-gomes-8178) [reply terrariumDatabase.py:731]: will be addressed in a follow-up PR

File: terrariumEngine.py [hunk 15]
```diff
@@ -2029,12 +2045,133 @@ def total_power_and_water_usage(self, force=False, background=False, thread_retu
...
+        """Load all feeders from database"""
+        from terrariumDatabase import Feeder as FeedersDB
+
+        self.feeders = {}
+
+        @orm.db_session
+        def _load():
+            for feeder_data in orm.select(f for f in FeedersDB):
+                try:
+                    feeder = terrariumFeeder(
```
[the remark this answers is not in the document]
Comment (ryan-gomes-8178) [reply terrariumEngine.py:2081]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/ryan-gomes-8178/PaludariumPI/pull/5#discussion_r2716274291)

File: gui/components/feeders/FeedersFormModal.svelte (hunk 3 above)
[the remark this answers is not in the document]
Comment (ryan-gomes-8178) [reply gui/components/feeders/FeedersFormModal.svelte:81]: already addressed in a different commit

File: terrariumEngine.py [hunk 16]
```diff
@@ -2029,12 +2045,133 @@ def total_power_and_water_usage(self, force=False, background=False, thread_retu
...
+
+                    schedule = feeder_db.schedule
+                    now = datetime.datetime.now()
+                    current_time = now.strftime("%H:%M")
+
+                    for feed_name, feed_config in schedule.items():
+                        if not feed_config.get("enabled", True):
+                            continue
+
+                        if feed_config.get("time") == current_time:
```
[the remark this answers is not in the document]
Comment (ryan-gomes-8178) [reply terrariumEngine.py]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/ryan-gomes-8178/PaludariumPI/pull/5#r2716274182)

[the remark this answers is not in the document]
Comment (ryan-gomes-8178) [reply terrariumEngine.py]: will be addressed in a follow-up PR
